framework/src/service/ServiceRegistrationBase.cpp

```diff
@@ -320,22 +306,25 @@ namespace cppmicroservices
         }
 
         {
-            auto l = d->Lock();
+            auto l = GetLocks();
             US_UNUSED(l);
 
-            d->bundle.reset();
-            d->dependents.clear();
-            d->service.reset();
-            d->prototypeServiceInstances.clear();
-            d->bundleServiceInstance.clear();
-            // increment the reference count, since "d->reference" was used originally
-            // to keep d alive.
-            ++d->ref;
+            d->coreInfo->bundle.reset();
+            d->coreInfo->dependents.clear();
+            d->coreInfo->service.reset();
+            d->coreInfo->prototypeServiceInstances.clear();
+            d->coreInfo->bundleServiceInstance.clear();
+
             d->reference = nullptr;
-            d->unregistering = false;
+            d->coreInfo->unregistering = false;
         }
     }
 
+    std::shared_ptr<LockSet> ServiceRegistrationBase::GetLocks() const{
```

Comment:
I'd like to see this function renamed to something more descriptive. this function acquires locks an RAII type object, but the function name does not convey any of this to the user.

Reply:
Changed it to `LockRegistration`... This gave me more insight into the fact that this was a call to lock the registration object generally. I'm not sure if this is clearer given that you could also say this sounds like `registration.Lock()`, which is only part of it. If anyone has naming ideas, I'm definitely open to that

https://github.com/CppMicroServices/CppMicroServices/pull/841/commits/5fc7802a40e3c35d2957b0562f56e73fb2744c23